test/assertions/enabled-test.js

```diff
@@ -0,0 +1,178 @@
+import chai, { expect } from 'chai';
+import sinonChai from 'sinon-chai';
+import FakeClient from '../stubs/fake-client';
+import sinon from 'sinon';
+import enabled from '../../src/assertions/enabled';
+import immediately from '../../src/chains/immediately';
+
+//Using real chai, because it would be too much effort to stub/mock everything
+chai.use(sinonChai);
+
+describe('enabled', () => {
+    let fakeClient;
+
+    beforeEach(() => {
+        fakeClient = new FakeClient();
+
+        fakeClient.isEnabled.throws('ArgumentError');
+        fakeClient.isEnabled.withArgs('.some-selector').returns(false);
+
+        chai.use((chai, utils) => enabled(fakeClient, chai, utils));
+        chai.use((chai, utils) => immediately(fakeClient, chai, utils));
+    });
+
+    afterEach(() => fakeClient.__resetStubs__());
+
+    describe('When in synchronous mode', () => {
+        describe('When not negated', () => {
+            beforeEach(() => {
+                fakeClient.isEnabled.withArgs('.some-selector').returns(true);
+
+                expect('.some-selector').to.be.enabled();
+            });
+
+            it('Should call `waitForEnabled` without `reverse`', () => {
+                expect(fakeClient.waitForEnabled)
+                    .to.have.been.calledWith('.some-selector', 0, undefined);
+            });
+
+            describe('When the element is still not enabled after the wait time', () => {
+                let testError;
+
+                beforeEach(() => {
+                    testError = 'Element still not enabled';
+
+                    fakeClient.waitForEnabled.throws(new Error(testError));
+                });
+
+                it('Should throw an exception', () => {
+                    expect(() => expect('.some-selector').to.be.enabled())
+                        .to.throw(testError);
+                });
+            });
+        });
+
+        describe('When negated', () => {
+            beforeEach(() => expect('.some-selector').to.not.be.enabled());
+
+            it('Should call `waitForEnabled` with `reverse` true', () => {
+                expect(fakeClient.waitForEnabled)
+                    .to.have.been.calledWith('.some-selector', 0, true);
+            });
+
+            describe('When the element is still enabled after the wait time', () => {
+                let testError;
+
+                beforeEach(() => {
+                    testError = 'Element still enabled';
+
+                    fakeClient.waitForEnabled.throws(new Error(testError));
+                });
+
+                it('Should throw an exception', () => {
```

Comment:
Is this test not a duplicate of the test below, "When the element is enabled > When the assertion is negated > Should throw an exception" ?

Reply:
(And similarly for the exception case above, under "When not negated")?  If they are duplicates, maybe we could remove these contexts, and move the "Should call `waitForEnabled`" tests into the contexts below?

Reply:
I think you are right with the two cases from "When not negated" and "When negated". I remove then.  The "When the element is enabled > When the assertion is negated > Should throw an exception" nevertheless I will leave it just as it is, since there are no mocks to the function `waitForEnabled` and test the real thing.